packages/yoastseo/src/markers/removeDuplicateMarks.js

```diff
@@ -1,12 +1,19 @@
-import { uniqBy } from "lodash-es";
+import { uniqBy, isUndefined } from "lodash-es";
 
 /**
- * Removes duplicate marks from an array
+ * Removes duplicate marks from an array.
+ * If the marks object have position information, however,
+ * we don't want to remove the duplicated objects with the same original strings.
+ *
+ * @param {Array} marks The marks to remove duplications from.
  *
- * @param {Array} marks The marks to remove duplications from
  * @returns {Array} A list of de-duplicated marks.
  */
 function removeDuplicateMarks( marks ) {
+	if ( marks.length === 0 || ! isUndefined( marks[ 0 ].hasPosition() ) ) {
+		return marks;
+	}
+
```

Comment:
Not removing duplicates is simpler than removing the duplicates by looking at the value of `mark.getPosition()`. In any case, with the way that we create the mark objects with the position, there won't be any duplicates with the same value of position information. So in this case, removing duplicates in this helper won't be necessary